sonarts-core/docs/se/symbolic-execution-design-notes.md

```diff
@@ -0,0 +1,20 @@
+# Symbolic Execution Design Notes for SonarTS
+
+* The execution tree is currently visited depth-first and it's not currently using a worklist approach
+
+* Rules are invoked at the end of the full visit of the exploded graph
+** During execution we collect all the program states for each program point
+** After execution the rule (or other callback) is called by providing the program point and the full set of program states
+
+* Literal values are represented as `LiteralSymbolicValue`, which contains the exact value.
+** We assume that once constraints are implemented the `LiteralSymbolicValue` will provide a constraint that corresponds to the value upon request.
```

Comment:
@inverno something wrong with formatting. Probably you should use indent, not 2 stars